lib/rules/no-useless-rename.js

```diff
@@ -97,27 +105,19 @@ module.exports = {
 
             for (const property of node.properties) {
 
-                /*
-                 * TODO: Remove after babel-eslint removes ExperimentalRestProperty
-                 * https://github.com/eslint/eslint/issues/12335
-                 */
-                if (property.type === "ExperimentalRestProperty") {
```

Comment:
Is this safe to remove? Based on https://github.com/babel/babel-eslint#breaking-changes-in-v11xx, the last version of `babel-eslint` before the move to `@babel/eslint-parser` still had this node type, and I'd bet a lot of projects still use `babel-eslint`.

Reply:
Sorry for not leaving a comment about this change! I think it should be okay with the new condition for an early exit below (`property.type !== "Property"` instead of `property.type === "RestElement"`)

Reply:
Ah! I'm following how that fits together now. Thanks for clarifying.

Reply:
There is a test case with `ExperimentalRestProperty`:

https://github.com/eslint/eslint/blob/c79b8a9cd7f9e3da2c6dd41dd6a6c4e80b5500ce/tests/lib/rules/no-useless-rename.js#L118-L125